packages/front-end/components/Settings/EditDataSource/DataSourceInlineEditIdentityJoins/AddEditIdentityJoinModal.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { FC, useMemo } from "react";
+import {
+  DataSourceInterfaceWithParams,
+  IdentityJoinQuery,
+} from "back-end/types/datasource";
+
+import { useForm } from "react-hook-form";
+import Modal from "../../../Modal";
+import MultiSelectField from "../../../Forms/MultiSelectField";
+import CodeTextArea from "../../../Forms/CodeTextArea";
+import { isDuplicateIdentityJoin } from "./utils";
+
+type AddEditIdentityJoinModalProps = {
+  identityJoin: IdentityJoinQuery | null;
+  dataSource: DataSourceInterfaceWithParams;
+  mode: "add" | "edit";
+  onSave: (identityJoin: IdentityJoinQuery) => void;
+  onCancel: () => void;
+};
+
+export const AddEditIdentityJoinModal: FC<AddEditIdentityJoinModalProps> = ({
+  identityJoin,
+  mode,
+  dataSource,
+  onCancel,
+  onSave,
+}) => {
+  const identityTypes = dataSource.settings.userIdTypes || [];
+  const existingIdentityJoins = dataSource.settings.queries.identityJoins || [];
+
+  const defaultQuery = useMemo(() => {
+    return (
+      "SELECT \n" +
+      identityTypes
+        .map(({ userIdType }) => `  ${userIdType} as ${userIdType}`)
+        .join(", \n") +
+      "\nFROM my_table"
+    );
+  }, [identityTypes]);
+
+  const form = useForm<IdentityJoinQuery>({
+    defaultValues: {
+      ids: identityJoin?.ids || [],
+      query: mode === "add" ? defaultQuery : identityJoin?.query || "",
+    },
+  });
+
+  const handleSubmit = form.handleSubmit(async (value) => {
+    onSave(value);
+
+    form.reset({
+      ids: [],
+      query: "",
+    });
+  });
+
+  // region Validation
+
+  const userEnteredIdentityJoinIds = form.watch("ids");
+  const userEnteredQuery = form.watch("query");
```

Comment:
If we're doing validation, should we also do some simple SQL validation?  The SQL needs to contain at a minimum (all case insensitive):
- The word `select`
- The word `from`
- All of the selected userIdTypes

Reply:
Also, we're doing a lot of duplicate `form.watch(...)` calls in this file.  Maybe it's better to just do them once here and reference the variables everywhere else.

Reply:
I've removed duplicate `form.watch` calls and used the existing variables.

Reply:
I would be hesitant to roll my own SQL validation. I'd opt for a library. If I were to implement the suggestion, users can put those words in any order and it would be considered valid even though it's not. Adding a SQL validation library can also probably help users write better queries so they can see the validation on the fly. 

I think it would also be good if we add SQL to this part of the app that we add it everywhere the user enters SQL. I've created a ticket for us to address this as a standalone item: https://github.com/growthbook/growthbook/issues/542

Let me know if you're ok with proceeding without the SQL validation added to the scope.